src/core/Akka.Remote/Transport/Streaming/NetworkStreamTransport.cs

```diff
@@ -0,0 +1,289 @@
+    {
+        public string Hostname { get; }
+
+        public int Port { get; }
+
+        public IPAddress BindIp { get; }
+
+        public int BindPort { get; }
+
+        public TimeSpan ConnectionTimeout { get; }
+
+        public int SendBufferSize { get; }
+
+        public int ReceiveBufferSize { get; }
+
+        public int Backlog { get; }
+
+        public bool TcpNoDelay { get; }
+
+        public bool TcpKeepAlive { get; }
+
+        public NetworkStreamTransportSettings(Config config)
+            : base(config)
+        {
+            Hostname = config.GetString("hostname");
+
+            if (string.IsNullOrEmpty(Hostname))
+                Hostname = "localhost";
+
+            Port = config.GetInt("port");
+
+            Hostname = config.GetString("hostname");
+            Port = config.GetInt("port");
+
+            string bindHostname = config.GetString("bind-hostname");
+
+            if (string.IsNullOrEmpty(bindHostname))
+                bindHostname = Hostname;
+
+            IPAddress ip;
+            if (Hostname.Equals("localhost", StringComparison.OrdinalIgnoreCase))
+            {
+                BindIp = IPAddress.Loopback;
+            }
+            else if (IPAddress.TryParse(bindHostname, out ip))
+            {
+                // The socket is using DualMode, if ipv6 is supported, listening on IPv6Any
+                // will work for any address (ipv4 and ipv6)
+                if (IPAddress.Any.Equals(ip) && Socket.OSSupportsIPv6)
+                    ip = IPAddress.IPv6Any;
+
+                BindIp = ip;
+            }
+            else
+            {
+                // If hostname is not localhost or an ip, bind to IPAny
+                BindIp = Socket.OSSupportsIPv6 ? IPAddress.IPv6Any : IPAddress.Any;
+            }
+
+            string bindPortString = config.GetString("bind-port");
+
+            BindPort = string.IsNullOrEmpty(bindPortString) ? Port : int.Parse(bindPortString);
+
+            ConnectionTimeout = config.GetTimeSpan("connection-timeout");
+            SendBufferSize = GetByteSize(config, "send-buffer-size");
+            ReceiveBufferSize = GetByteSize(config, "receive-buffer-size");
+            
+            Backlog = config.GetInt("backlog");
+            TcpNoDelay = config.GetBoolean("tcp-nodelay");
+            TcpKeepAlive = config.GetBoolean("tcp-keepalive");
+        }
+
+        public void ConfigureSocket(Socket s)
+        {
+            s.SendBufferSize = SendBufferSize;
+            s.ReceiveBufferSize = ReceiveBufferSize;
+            s.NoDelay = TcpNoDelay;
+
+            if (TcpKeepAlive)
+                s.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.KeepAlive, 1);
+        }
+
+        public IPEndPoint GetBindEndPoint()
+        {
+            return new IPEndPoint(BindIp, BindPort);
+        }
+    }
+
+    public class NetworkStreamTransport : StreamTransport
+    {
+        public const string ProtocolName = "tcp";
```

Comment:
Whoops, nevermind I stand corrected - Helios / Netty only do this because they support UDP and TCP. This is fine otherwise.